### ISSUE TITLE
Inconsistent simulation results

### ISSUE THREAD

**Issue by macumber** [NREL/OpenStudio-resources#1]
@kbenne sometimes running the same script multiple times produces different results.  Initial culprits are:

ModelSimulationFixture.baseboard_heater_rb

ModelSimulationFixture.air_terminals_rb

**Comment by jmarrec** [comment]
`availability_managers.rb`, I ran the test 10 times in a row with the same machine using the same OS version (current develop), parsing total_site_energy_kbtu each time:

```
{0: 4065225.54170863,
 1: 3399497.7527429624,
 2: 4071405.309333074,
 3: 3187215.152306389,
 4: 3394019.3697875515,
 5: 3252206.972246273,
 6: 3398966.9751555873,
 7: 4431499.989682509,
 8: 4205701.517110268,
 9: 4071367.396648261}
```

**Comment by jmarrec** [comment]
this is part of the problem here: https://github.com/jmarrec/OpenStudio-resources/blob/develop/model/simulationtests/availability_managers.rb#L31

the zones and systems aren't in the same order in the file, so getting by index will assign different AVMs to the systems on subsequent runs.

I changed it to test by system name rather than just index, and I ran multiple time the test, the difference aren't as huge as before, but they are still there, so there's more to it.

**Comment by jmarrec** [comment]
I fixed most of these problems (will push soon), they all come from the fact that the `lib/baseline_model` stuff will create an OSM with zones not in the same order, and the tests themselves use `zones = model.getThermalZones` then `zones[i]`

So instead, I rely on the proper naming conventions for the thermal zones, by doing `zones = model.getThermalZones.sort_by{|z| z.name.to_s}`. This allows us to ensure we always create a system for the same zone.

The availability_manager had the same problem, it just wasn't sufficient to sort on AirLoopHVACs name, so I sort the airloopHVACs by the attached thermal zone names.

There are two tests that still have a bit of a stability issue, `surface_properties.rb` has about 1% difference, `fan_on_off` has 0.2%, so there's probably something else underneath, this time it may be convergence issues.

**Comment by jmarrec** [comment]
`fan_on_off`:

```
   ************* Beginning Simulation
   ** Warning ** CalcDoe2DXCoil: Coil:Cooling:DX:SingleSpeed "COIL COOLING DX SINGLE SPEED 3" - Air-cooled condenser inlet dry-bulb temperature below 0 C. Outdoor dry-bulb temperature = -7.80
   **   ~~~   **  ... Occurrence info = RUN PERIOD 1, 01/05 11:48 - 11:49
   **   ~~~   ** ... Operation at low ambient temperatures may require special performance curves.
   ** Warning ** CalcDoe2DXCoil: Coil:Cooling:DX:SingleSpeed "COIL COOLING DX SINGLE SPEED 1" - Air-cooled condenser inlet dry-bulb temperature below 0 C. Outdoor dry-bulb temperature = -3.75
   **   ~~~   **  ... Occurrence info = RUN PERIOD 1, 02/16 14:10 - 14:15
   **   ~~~   ** ... Operation at low ambient temperatures may require special performance curves.
   ** Warning ** Part-load ratio heating control failed in fan coil unit ZONE HVAC FOUR PIPE FAN COIL 1
   **   ~~~   **   Bad hot part-load ratio limits
   **   ~~~   ** ..Part-load ratio set to 0 Environment=RUN PERIOD 1, at Simulation time=04/11 15:38 - 15:45
   ** Warning ** Plant Loop: CONDENSER WATER LOOP Demand Side is storing excess heat the majority of the time.
   ** Warning ** Plant Loop: CONDENSER WATER LOOP Supply Side is storing excess heat the majority of the time.

[...]

   ************* ===== Recurring Error Summary =====
   ************* The following recurring error messages occurred.
   *************
   *************  ** Warning ** CalcDoe2DXCoil: Coil:Cooling:DX:SingleSpeed="COIL COOLING DX SINGLE SPEED 3" - Low condenser dry-bulb temperature error continues...
   *************  **   ~~~   **   This error occurred 423 total times;
   *************  **   ~~~   **   during Warmup 0 times;
   *************  **   ~~~   **   during Sizing 0 times.
   *************  **   ~~~   **   Max=-0.3 [C]  Min=-15. [C]
   *************
   *************  ** Warning ** CalcDoe2DXCoil: Coil:Cooling:DX:SingleSpeed="COIL COOLING DX SINGLE SPEED 1" - Low condenser dry-bulb temperature error continues...
   *************  **   ~~~   **   This error occurred 46 total times;
   *************  **   ~~~   **   during Warmup 0 times;
   *************  **   ~~~   **   during Sizing 0 times.
   *************  **   ~~~   **   Max=-0.275 [C]  Min=-4.4 [C]
   *************
   *************  ** Warning ** Part-load ratio heating control failed in fan coil unit ZONE HVAC FOUR PIPE FAN COIL 1
   *************  **   ~~~   **   This error occurred 409 total times;
   *************  **   ~~~   **   during Warmup 0 times;
   *************  **   ~~~   **   during Sizing 0 times.

```

**Comment by jmarrec** [comment]
ok I think I fix the last instability in #42.

I ran all tests 5 times in a row using current OpenStudio develop, and there is no difference at all, at least for tests that pass (there are a handlful that are failing, mostly due to new E+ 8.9.0 changes)

Here are all differences:

![image](https://user-images.githubusercontent.com/5479063/37917728-7a376040-311f-11e8-822a-415be4d5cd7a.png)

So evaporative_cooling.osm (note that is is an **OSM one, not ruby**, so that indicates convergence issues) is the only one with 0.0032% difference on one run...

I think we can close this issue once #42 is merged in.

**Comment by jmarrec** [comment]
Fixed in #42
